tests/unit/test_network_session.py

```diff
@@ -219,3 +219,11 @@ def warning(self, *args, **kwargs):
         actual_level, actual_message = log_records[0]
         assert actual_level == 'WARNING'
         assert 'is not a trusted or secure host' in actual_message
+
+    def test_extra_headers(self):
```

Comment:
I would add a `test_bogus_extra_header` and see that am exception is raised

Reply:
done